### PR TITLE
Func

### PR DESCRIPTION
**Pull request by kbrownc**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,34 +1,30 @@
...
-  onReset = () => {
-    this.setState({
-      roll: null,
-      message: 'Roll again',
-      optMessage: 'Kim',
-      score: 0,
-      position: -1,
-      board: newBoard.slice(),
-    });
+  const onReset = () => {
```
Comment (Luke-Michael-Brown) [inline src/App.js]: This function should be wrapped in a useCallback

File: src/App.js [hunk 2]
```diff
@@ -1,34 +1,30 @@
...
+    setOptMessage("Kim");
+    setScore(0);
+    setPosition(-1);
+    setBoard(newBoard.slice());
   };
   // roll dice (roll button)
-  onRoll = () => {
-    this.setState(oldState => {
-      let workBoard = oldState.board.slice();
+  const onRoll = () => {
```
Comment (Luke-Michael-Brown) [inline src/App.js]: This function should be wrapped in a useCallback

File: src/App.js [hunk 3]
```diff
@@ -1,34 +1,30 @@
...
+      let workBoard = board.slice();
       const randomNumber = Math.floor(Math.random() * 6) + 1;
       let scoreAdj = 1;
-      let newPosition = Math.min(oldState.position + randomNumber, workBoard.length - 1);
```
Comment (Luke-Michael-Brown) [inline src/App.js:31]: We still need to refer to oldState. You can do the same thing we useState hooks as you did in class setState.
e.g.
```
const [foo, setFoo] = useState(0);
...
setFoo((oldFoo) => {
  return oldFoo + 1;
});
...
```

File: src/App.js [hunk 4]
```diff
@@ -58,52 +54,50 @@ class App extends React.Component {
...
-        board: workBoard,
-        roll: randomNumber,
-        message: message,
-        optMessage: optMessage,
-        score: oldState.score + scoreAdj,
-        position: newPosition,
-      };
-    });
+ //     return {
+        setRoll(randomNumber)
```
Comment (Luke-Michael-Brown) [inline src/App.js]: Whenever you find yourself doing a bunch of setState's in a row this means you should probably connect the state.
e.g.
```
const [{ roll, position }, setGameState] = useState({
  roll: 0.
  position: -1,
});
...
setGameState((oldGameState) => {
  const randomNumber = Math.floor(Math.random() * 6) + 1;
  const newPosition = Math.min(oldGameState.position + randomNumber, workBoard.length - 1);
  return {
    roll: randomNumber,
    position: newPosition
  };
});
```

**Comment by Luke-Michael-Brown** [comment]
Don't forget to run prettier :)

File: src/App.js (hunk 1 above)
Comment (kbrownc) [reply src/App.js]: Ok....it's been added....but it doesn't make sense without any dewpendancies.

File: src/App.js (hunk 2 above)
Comment (kbrownc) [reply src/App.js]: Ok......done.....but the dependencies the warning message told me to use doesn't make sense.

File: src/App.js (hunk 4 above)
Comment (kbrownc) [reply src/App.js]: OK....I tried this...but then newPosition is no longer available for the rest of the code.

File: src/App.js (hunk 3 above)
Comment (kbrownc) [reply src/App.js:31]: I added this...but am not sure where else I should be using prevState.

File: src/App.js (hunk 4 above)
Comment (kbrownc) [reply src/App.js]: And I fixed up the message and optMessage properties to make them different from what I used to set state.

File: src/App.js (hunk 2 above)
Comment (kbrownc) [reply src/App.js]: I've only got 2 dependancies

File: src/App.js (hunk 4 above)
Comment (kbrownc) [reply src/App.js]: And then I decided the put all of the fields in the setState.

File: src/App.js (hunk 1 above)
Comment (Luke-Michael-Brown) [reply src/App.js]: it's still needed without dependencies, it'll make sure the function reference doesn't change.

File: src/App.js [hunk 5]
```diff
@@ -49,70 +52,69 @@ class App extends React.Component {
...
-        score: oldState.score + scoreAdj,
         position: newPosition,
+        message: workMessage,
+        optMessage: workOptMessage,
+        score: prevGameState.score + scoreAdj,
+        board: workBoard,
       };
     });
-  };
+  }, [board]);
```
Comment (Luke-Michael-Brown) [inline src/App.js]: You shouldn't need to use board as a dependency you should be able to use `prevGameState.board` in the function instead and then remove this.

File: src/App.js (hunk 1 above)
Comment (kbrownc) [reply src/App.js]: Function reference?

File: src/App.js (hunk 5 above)
Comment (kbrownc) [reply src/App.js]: Done.

File: src/App.js (hunk 1 above)
Comment (Luke-Michael-Brown) [reply src/App.js]: I can explain in more detail sometime in person, it's a little too complicated for a PR comment 😅  But nothing you can't handle...for now just trust me aha

File: src/App.js [hunk 6]
```diff
@@ -1,43 +1,46 @@
...
-  onReset = () => {
-    this.setState({
-      roll: null,
-      message: 'Roll again',
-      optMessage: 'Kim',
-      score: 0,
-      position: -1,
-      board: newBoard.slice(),
+  const onReset = useCallback(() => {
+    setGameState(prevGameState => {
```
Comment (Luke-Michael-Brown) [inline src/App.js]: This actually doesn't need the `prevGameState` since you don't use it. You can just do the object directly.

**Comment by Luke-Michael-Brown** [comment]
After updating `onReset`'s `prevGameState` and running prettier you can merge this 👍

File: src/App.js (hunk 1 above)
Comment (kbrownc) [reply src/App.js]: OK

File: src/App.js (hunk 6 above)
Comment (kbrownc) [reply src/App.js]: I presume you mean replace prevGameState   with '()'

**Pull request merged**